File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0) return 0;
        Set<Integer> set = new HashSet<>();
        for(int n : nums) {
            set.add(n);
        }

        int longest = 1;
        for(int st : set) {
            if(!set.contains(st-1)){
                int cnt = 1;
                int temp = st;
                while(set.contains(temp+1)) {
                    temp++;
                    cnt++;
                }
                longest = Math.max(cnt, longest);
            }
        }
        return longest;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        vals = set(nums)
        longest = 1

        for val in vals:
            if val-1 not in vals:
                temp = val
                ctn = 1
                while temp + 1 in vals:
                    ctn += 1
                    temp += 1
                longest = max(longest, ctn)
        return longest